### PR TITLE
Tabular: Optimized LightGBM and XGBoost

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
#1324 

Special thanks to @ShaohongBai for mentioning the unexpected training speed with F1 metric!

*Description of changes:*

- Dramatically sped up LightGBM training by removing the evaluation of training data at every iteration.
  - 20x+ speedup in many situations on large datasets with expensive evaluation metrics (`F1`).
- Dramatically sped up XGBoost during refit_full by removing the evaluation of training data at every iteration.
  - 20x+ speedup in many situations on large datasets with expensive evaluation metrics (`F1`).
- Speedup for all datasets regardless of metric. Using fast metrics on large datasets can still provide 50% or more speedup.
- Updated LightGBM model to avoid warning introduced in LightGBM v3.3
- Updated XGBoost model to avoid warning introduced in XGBoost v1.3
- Updated LightGBM minimum version due to breaking LightGBM API change to resolve warning in LightGBM v3.3.

TODO:

- [x] Benchmark on AutoMLBenchmark

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1344-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1344/1/index.html

**Comment by Innixma** [comment]
Benchmark Results:

LightGBM even when using fast metrics has sped up significantly on large datasets.

(Binary) sf-police-incidents: 508s -> 334s
(Binary) Higgs: 310s -> 221s
(Multiclass) covertype: 967s -> 732s
(Multiclass) KDDCup99: 154s -> 116s
(Regression) Airlines_DepDelay_10M: 808s -> 752s

These speedups would likely be far more dramatic with unoptimized metrics, such as `f1`.

AutoMLBenchmark comparison shows that faster LightGBM translates to better model quality (52% winrate):

```
AutoGluon_bestquality_4h8c_2021_10_09_cat VS all
                                   framework   winrate    >    <     =
0  AutoGluon_bestquality_4h8c_2021_10_16_lgb  0.521401  493  449    86
```

**Comment by szha** [comment]
Job PR-1344-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1344/3/index.html
